uniffi/src/bindings/kotlin/gen_kotlin.rs

```diff
@@ -63,6 +64,20 @@ mod filters {
             TypeReference::Record(_) => "RustBuffer.ByValue".to_string(),
             TypeReference::Optional(_) => "RustBuffer.ByValue".to_string(),
             TypeReference::Object(_) => "Long".to_string(),
+            TypeReference::String => "String".to_string(),
```

Comment:
I was wondering, do we need the same workaround as for Python and use `Pointer` here? I remember @thomcc saying something about `String` not being safe for JNA, and if I'm reading our [docs](https://github.com/mozilla/application-services/blob/main/docs/howtos/exposing-rust-components-to-kotlin.md) right, they say to use `Pointer` for Rust strings, but maybe this is different? (And if we could use `Pointer`, would that let us merge `String` and `CString`?)

Reply:
Summoned!

- For strings passed as arguments from Kotlin to Rust, using `String` on the JVM side (and e.g. `FfiStr<'_>` on Rust) is correct. 
- For strings returned by rust, these must be freed later, and so we need to use a Pointer in order to maintain the same address for when it's later passed to our string freeing function

So, yes, depending on the case, we need to use Pointer for strings otherwise we'll leak them.

Reply:
Oh, cool, Tarik's PR is right as-is and I learned something new (it's OK to use `String` for arguments from Kotlin to Rust, just not for returns from Rust to Kotlin).